### ISSUE TITLE
Lit Support: Route case assignments for inactive attorney profiles to the general work queue 

### ISSUE THREAD

**Issue by jimruggiero** [department-of-veterans-affairs/caseflow#12284]
### User or job story
User story: When a Motions Attorney, who was originally working a given case no longer works at BVA or in Lit Support, I want the the case returned to the team queue, so that it can be re-assigned to a different attorney.

### Acceptance criteria
- This work is behind a feature toggle: "review_motion_to_vacate".
- When motions attorney profiles are marked as inactive in Caseflow, all cases assigned to that attorney are automatically moved to the respective generic team queue(s).
- Cases moved into generic team queues are visible and can by reassigned by team leads.
- Once reassigned, cases are visible by the newly assigned attorney. 
- Newly assigned attorneys can see reassigned cases in their work queues.
- A screenshot or .gif of what is coded and merged to Production is attached to this issue prior to it being closed.

### Release notes
_Write what should be included in release notes (Caseflow uses Headway), updated when the story is built, before it's deployed._

Caseflow now supports the ability for cases assigned to attorneys no longer working at The Board or in Litigation Support, to be routed to a teams' general work queue for reassignment to a new attorney.

#### Designs
https://www.figma.com/file/VtOG0hibbyJ6T5UcYnn5EP/Motions-to-Vacate_Phase-II_Epic-8547?node-id=0%3A1

#### Background/context
This story specifically drives value for BVA team leads. It allows them to quickly reassign cases to prevent unnecessary delays in the process for veterans and ensures productivity metrics are met by their respective teams.

#### Success criteria
See ACs

#### Technical notes
- One approach would be to do this through an after save hook on users.
- Close the task associated with the inactive attorney and open a new task
- User tasks should all have parent organizations tasks

#### Open questions
- Does this need to support more than Lit Support (i.e. when returning tasks to generic queues, would there be different queues for different teams [motions v decisions attorney teams])?
- Can we extend this to judge reassignment flows detailed in other tickets?

#### Other notes
- Newly assigned attorneys can see the prior (inactive) attorney assignment and the new assignment of the case (to them) in Case Timeline - out of scope for this epic

#### Resources/other links
Related PR for reassigning tasks for inactive users: https://github.com/department-of-veterans-affairs/caseflow/pull/12483

**Comment by jimruggiero** [comment]
Please reference work that Echo is doing to hook into inactive user/general queue task assignments (see ticket XXXXX).

**Comment by jcq** [comment]
Had a convo w/ Echo re MTV task tree.

Takeaways:
- Our general task structure is fine
- Suggested we replace `AbstractMotionToVacateTask` with `JudgeSignMotionToVacateTask` in the grant/partial cases, as that follows how other judge tasks are handled. This would create the `JudgeSignMotionToVacateTask` right away and put it on hold (judges don't currently have an "on-hold" tab, but they will once the PR for that lands), (re)opening it up for the judge once the attorney task is complete.
- We should create tasks assigned to LitSupport org for the `denied` and `dismissed` tasks in addition to the ones assigned to the individual user, as that will serve as fallback in case user goes inactive
- For `VacateAndDeNovoTask` and `StraightVacateAndReadjudicationTask`, we need to decide best way to account for the assigned user going inactive. We could create some sort of process by which one of the earlier judge tasks (`AddressMotionToVacate`) gets reopened somehow to allow recreation of these tasks by selecting a new attorney, or we could just automatically create a second (parent) task for each of these that are assigned to the judge's team as an org.

**Comment by jimruggiero** [comment]
Hey JC,

Thanks for all of this due diligence. Bullets 1-3 - great news!

Regarding the last bullet (De Novo and Straight Vacate), I'm wondering if
shifting the lens through which we're defining the problem space would
help. If these tasks were going to be shown in Timeline where, in some
ideal future state we are truly showing each incremental step (i.e. full
audit history for all appeals, issues, decisions) in the history (including
judge task assign/reassign due to inactive and attorney task
assign/reassign due to inactive; knowing that multiple reassignments per
role are possible), would surfacing those incremental tasks (with accurate
context in the Timeline UI) influence which path or solution options you
and the team think would be best? @sjones@tistatech.com
<sjones@tistatech.com>  I know you are familiar with Timeline code. Any
preference or recommendations you would offer JC?

As much as I don't want to over-think this or derail by injecting
unnecessary/irrelevant scope, I do think we should briefly think big
picture implications to (hopefully minimize refactoring work down the road
and long-term scalability of the approach, given the other post-decision
motions workflows will someday leverage or tie into this functionality. We
should document our decisions, rationale and solution limitations in the
story for this one (brief bullets), as it's likely that other teams will
build future post-decisional motions workflows in Caseflow.
